packages/core/src/util/cache.js

```diff
@@ -40,19 +67,37 @@ export function lruCache({
     }
   }
 
+  function get(key) {
+    const entry = cache.get(key);
+    if (entry) {
+      entry.last = performance.now();
+      return entry.value;
+    }
+  }
+
+  function set(key, value) {
+    cache.set(key, { last: performance.now(), value });
+    if (cache.size > max) requestIdle(evict);
+    return value;
+  }
+
   return {
-    get(key) {
-      const entry = cache.get(key);
-      if (entry) {
-        entry.last = performance.now();
-        return entry.value;
-      }
+    get,
+    set,
+    clear() { cache = new Map; },
+    export() {
+      return Array.from(cache).reduce(
+        (acc, [key, entry]) => {
+          acc.set(key, tableToIPC(entry.value, { format: 'stream' }));
+          return acc;
+        },
+        new Map()
+      );
     },
-    set(key, value) {
-      cache.set(key, { last: performance.now(), value });
-      if (cache.size > max) requestIdle(evict);
-      return value;
-    },
-    clear() { cache = new Map; }
+    import(data) {
+      Array.from(data).forEach(
```

Comment:
Here is an alternative. Note that we need to use `decodeIPC` (from the util folder) to ensure consistent Arrow parsing relative to the database connectors.

```js
for (const [key, entry] of data) {
  set(key, decodeIPC(entry.value));
}
```